apps/dashboard/src/modules/interest-group/modals/create-interest-group-modal.tsx

```diff
@@ -0,0 +1,26 @@
+import { ContextModalProps, modals } from "@mantine/modals"
+import { FC } from "react"
+import { useCreateInterestGroupMutation } from "../mutations/use-create-interest-group-mutation"
+import { useInterestGroupWriteForm } from "src/app/(dashboard)/interest-group/write-form"
+
+export const CreateInterestGroupModal: FC<ContextModalProps> = ({ context, id }) => {
+  const close = () => context.closeModal(id)
+  const create = useCreateInterestGroupMutation()
+  const FormComponent = useInterestGroupWriteForm({
+    onSubmit: (data) => {
+      create.mutate({
+        ...data,
+      })
+      close()
```

Comment:
If this fails the user won't get any feedback.. But this is the case for other create modals as well. So I think we should go over everything at a later point and fix it. But if you want it would be nice to get some feedback on fail.

Reply:
It will still get an error if the mutation fails so I don't see this as a major issue. We can add an event listener to the form jsx component but I don't see it as necessary now